code/lib/preview-api/src/modules/store/csf/prepareStory.ts

```diff
@@ -98,26 +99,57 @@ export function prepareStory<TRenderer extends Renderer>(
     storyAnnotations?.render ||
     componentAnnotations.render ||
     projectAnnotations.render;
-  if (!render) throw new Error(`No render function available for storyId '${id}'`);
 
   const decoratedStoryFn = applyHooks<TRenderer>(applyDecorators)(undecoratedStoryFn, decorators);
   const unboundStoryFn = (context: StoryContext<TRenderer>) => decoratedStoryFn(context);
 
   const playFunction = storyAnnotations?.play ?? componentAnnotations?.play;
 
+  const mountUsed = mountDestructured(playFunction);
+
+  if (!render && !mountUsed) {
+    throw new Error(`No render function available for storyId '${id}'`);
```

Comment:
Should this be a named error?

Reply:
added a TODO, I didn't add this line, I don't have the mental headspace right now to add the right named error